caliban/hooks/git_commit_hooks.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Git commit prerun and prebuild hooks.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+from typing import Any, Dict, List, NamedTuple, NewType, Optional, Union
+
+import sys
+from absl import logging
+import docker
+from git import Repo
+
+
+def get_hash_from_image(container_id):
+  """Load git hash from docker image label."""
+  client = docker.from_env()
+  labels = client.images.get(container_id).labels
+  try:
+    return labels['commit']
+  except KeyError as e:
+    logging.error(f'Docker image {container_id} does not contain a git commit label.')
+    raise e
+
+    
+def git_commit_prebuild_hook():
+  """Git commit hash prebuild hook."""
+  logging.info('Running git-cleanliness prebuild hook.')
+
+  repo = Repo('.') # TODO deal with this
+
+  # check repo is clean
+  if len(repo.untracked_files) != 0:
```

Comment:
This is also fixed in the new, python-ed version.  Rather than using `subprocess`, we now require that hooks be python functions which return, in case of success, a dictionary like `{'Succeeded': True, 'Data': {'commit': 'asdfa'}}`, and in the case of failure, a dictionary like `{'Succeeded': False, 'Error': 'Error message'}`